telecine/src/main/java/com/jakewharton/telecine/RecordingSession.java

```diff
@@ -156,17 +158,53 @@ private void startRecording() {
     int sizePercentage = videoSizePercentage.get();
     Timber.d("Video size: %s%%", sizePercentage);
 
-    DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
-    int displayHeight = displayMetrics.heightPixels * sizePercentage / 100;
-    int displayWidth = displayMetrics.widthPixels * sizePercentage / 100;
-    int displayDpi = displayMetrics.densityDpi;
+    DisplayMetrics displayMetrics = new DisplayMetrics();
+    WindowManager wm = (WindowManager) context.getSystemService(WINDOW_SERVICE);
+    wm.getDefaultDisplay().getRealMetrics(displayMetrics);
+
+    Configuration configuration = context.getResources().getConfiguration();
+
+    int maximumWidth;
+    int maximumHeight;
+
+    // Get the best camera profile available. We assume MediaRecorder supports the highest.
+    CamcorderProfile camcorderProfile = CamcorderProfile.get(CamcorderProfile.QUALITY_HIGH);
+    if (camcorderProfile != null) {
+      boolean isLandscape = configuration.orientation == Configuration.ORIENTATION_LANDSCAPE;
+      int frameWidth = camcorderProfile.videoFrameWidth;
+      int frameHeight = camcorderProfile.videoFrameHeight;
+      maximumWidth = isLandscape ? frameWidth : frameHeight;
+      maximumHeight = isLandscape ? frameHeight : frameWidth;
+
+      // Check if video recording quality is higher than the screen resolution
+      if(maximumWidth > displayMetrics.widthPixels && maximumHeight > displayMetrics.heightPixels) {
+        maximumWidth = displayMetrics.widthPixels;
+        maximumHeight = displayMetrics.heightPixels;
+      } else {
+        // If not then we calculate new width or height to preserve aspect ratio
+        if(isLandscape) {
+          maximumWidth = displayMetrics.widthPixels * maximumHeight / displayMetrics.heightPixels;
```

Comment:
I don't think this logic is suitable. You can't guarantee that this is the side that needs adjusted. Some phones might be wider than the camera's capability and some might be taller than it.